packages/x-components/src/x-modules/tagging/components/__tests__/tagging.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue from 'vue';
+import { Component, Prop, Provide } from 'vue-property-decorator';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
+import { XComponentBusAPI } from '../../../../plugins/x-plugin.types';
+import { baseSnippetConfig } from '../../../../views/base-config';
+import { SnippetConfig } from '../../../../x-installer/api/api.types';
+import Tagging from '../../../tagging/components/tagging.vue';
+
+@Component({
+  template: `<div><slot/></div>`
+})
+class Provider extends Vue {
+  @Prop({ default: baseSnippetConfig })
+  @Provide()
+  public snippetConfig!: SnippetConfig;
+}
+
+function renderTagging({
+  template = `<Provider :snippetConfig="snippetConfig"><Tagging :consent="consent"/></Provider>`,
+  consent = undefined,
+  snippetConfig = baseSnippetConfig
+}: RenderTaggingOptions = {}): RenderTaggingAPI {
+  const [, localVue] = installNewXPlugin();
+
+  const wrapper = mount(
+    {
+      components: {
+        Tagging,
+        Provider
+      },
+      props: ['consent', 'snippetConfig'],
+      template
+    },
+    {
+      propsData: {
+        consent,
+        snippetConfig
+      },
+      localVue
+    }
+  );
+
+  return {
+    wrapper: wrapper.findComponent(Tagging),
+    on: wrapper.vm.$x.on.bind(wrapper.vm.$x)
+  };
+}
+
+describe('testing Tagging component', () => {
+  it('is an XComponent which has an XModule', () => {
+    const { wrapper } = renderTagging();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('tagging');
+  });
+
+  it('emits ConsentProvided when the consent is set and send it by prop', () => {
+    const { on } = renderTagging({ consent: true });
+    const eventSpy = jest.fn();
+
+    on('ConsentProvided').subscribe(eventSpy);
+    expect(eventSpy).toHaveBeenNthCalledWith(1, true);
+  });
+
+  it("doesn't emit ConsentProvided when the consent is undefined send it by prop", () => {
+    const { on } = renderTagging();
+    const eventSpy = jest.fn();
+
+    on('ConsentProvided').subscribe(eventSpy);
+    expect(eventSpy).not.toHaveBeenCalled();
+  });
+
+  // eslint-disable-next-line max-len
+  it('emits ConsentProvided when the consent is provided and send it using the snippet config', async () => {
+    const snippet = Vue.observable({
+      ...baseSnippetConfig,
+      consent: false
+    } as SnippetConfig & { consent?: boolean });
```

Comment:
Why do you need that ugly cast?

```suggestion
const snippet = Vue.observable<SnippetConfig>({
      ...baseSnippetConfig,
      consent: false
    });
```